### PR TITLE
Updates on CAM Module

### PR DESCRIPTION
**Pull request by Andrei-Matibag**
**Overview**

- Updated UI of Admin Creation.
- Creation of UI and Back-end of Committee Table.
- Backend of Admin Creation.
- Normalize File Path of Committee Table.
- Removed Session Manager.

**Additional Information**

- There is a problem in the footer in Committee Table.
- Responsiveness and Back-end of Account Details is on progress.
- Once the admin account has been created, there is an issue with login. 
- Instead of the generated password, It uses the hash password saved in the database.
- Validation in input field on Admin Creation is on progress.
- The auto-generated password hasn't been sent to the email yet.

### CODE REVIEW

**Comment by Andrei-Matibag** [comment]
**Additional**

- manage-members.php
-This code allows only the verified accounts of committee/admin member to show on table and it will only show the accounts that has the role of committee member and admin member only.

File: src/admin-creation.php [hunk 1]
```diff
@@ -1,119 +1,147 @@
...
-
-<body>
-
-	<!---------- SIDEBAR + HEADER START ------------>
-	<?php include_once __DIR__ . '/includes/components/sidebar.php'; ?>
-	<!---------- SIDEBAR + HEADER END ------------>
-
+<?php
+require_once 'includes/classes/db-connector.php';
+require_once 'includes/session-handler.php';
```
Comment (C-Ivan-Bandilla) [inline src/admin-creation.php:3]: Kindly use the `normalizeFilepath` function for **includes** and **requires**

Here is the sample
image

the function is available if you include the file utility `class include_once str_replace('/', DIRECTORY_SEPARATOR, 'includes/classes/file-utils.php');` _**file path may vary**_

The purpose is that we are not sure of what operating system our website would be hosted. The operating system varies on directory separators read more [here](https://en.wikipedia.org/wiki/Path_(computing)#:~:text=paths%20are%20appended.-,Representations%20of%20paths%20by%20operating%20system%20and%20shell,-%5Bedit%5D)
This function appropriately assigns the correct directory separator for the current OS. It's better to be safe thank you.

**Review by C-Ivan-Bandilla — CHANGES_REQUESTED** [review]

File: src/submission_handlers/manage-members.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+$conn = DatabaseConnection::connect();
+
+$queryExecutor = new QueryExecutor($conn);
+$query = "SELECT * FROM voter WHERE status = 'For Verification'";
```
Comment (Carl-Tabuso) [inline src/submission_handlers/manage-members.php:6]: Use prepared statement. Your query is susceptible to SQL injection.

**Review by Carl-Tabuso — CHANGES_REQUESTED** [review]
Check comments and do the necessary changes.

File: src/admin-creation.php [hunk 3]
```diff
@@ -1,119 +1,148 @@
...
-	<title>Admin Creation</title>
+require_once FileUtils::normalizeFilePath('includes/classes/db-connector.php');
+require_once FileUtils::normalizeFilePath('includes/session-handler.php');
+require_once FileUtils::normalizeFilePath('includes/classes/session-manager.php');
+require_once FileUtils::normalizeFilePath('includes/classes/query-handler.php');
 
-	<!-- Icons -->
-	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.2.1/css/all.min.css" />
-	<script src="https://cdn.jsdelivr.net/npm/feather-icons/dist/feather.min.js"></script>
+if (isset($_SESSION['voter_id'])) {
```
Comment (Carl-Tabuso) [inline src/admin-creation.php:9]: You're not checking if the user role is set to Committee Member.

File: src/manage-committee.php [hunk 4]
```diff
@@ -0,0 +1,334 @@
+<?php
+require_once 'includes/classes/db-connector.php';
+require_once 'includes/session-handler.php';
+require_once 'includes/classes/session-manager.php';
```
Comment (Carl-Tabuso) [inline src/manage-committee.php:4]: You only require the session manager if your page is static. Remove this.

File: src/manage-committee.php [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+require_once 'includes/classes/query-handler.php';
+
+if (isset($_SESSION['voter_id'])) {
```
Comment (Carl-Tabuso) [inline src/manage-committee.php:7]: You don't check for the role of the user.

**Review by Carl-Tabuso — CHANGES_REQUESTED** [review]
Your code is prone to URL directory listing. And the same as before, some of your files that require files from another folder or directory are not normalized. See comments below.

**Comment by Andrei-Matibag** [comment]
The changes has been made

**Review by Carl-Tabuso — CHANGES_REQUESTED** [review]
This is such a redundancy. You can check the codes of admin dashboard on how it implements role checking.

image

**Review by Carl-Tabuso — APPROVED** [review]
We'll get back on this issues you are yet to address in  your next PR.

**Pull request merged**